dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/QaServiceTest.kt

```diff
@@ -0,0 +1,47 @@
+package org.dataland.e2etests.tests
+
+import org.awaitility.Awaitility.await
+import org.dataland.datalandbackend.openApiClient.model.CompanyAssociatedDataEuTaxonomyDataForNonFinancials
+import org.dataland.datalandbackend.openApiClient.model.QAStatus as BackendQaStatus
+import org.dataland.datalandqaservice.openApiClient.model.QAStatus as QaServiceQaStatus
+import org.dataland.e2etests.auth.TechnicalUser
+import org.dataland.e2etests.utils.ApiAccessor
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import java.util.concurrent.TimeUnit
+
+class QaServiceTest {
+    private val apiAccessor = ApiAccessor()
+
+    private val testDataEuTaxonomyNonFinancials = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials
+        .getTData(1).first()
+
+    private val testCompanyInformation = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials
+        .getCompanyInformationWithoutIdentifiers(1).first()
+
+    @Test
+    fun `post dummy data, qa it and check the qa status changes`() {
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+        val storedCompanyInfos = apiAccessor.companyDataControllerApi.postCompany(testCompanyInformation)
+        val companyAssociatedEuTaxonomyNonFinancialsData =
+            CompanyAssociatedDataEuTaxonomyDataForNonFinancials(
+                storedCompanyInfos.companyId,
+                "",
+                testDataEuTaxonomyNonFinancials,
+            )
+
+        val dataId = apiAccessor.dataControllerApiForEuTaxonomyNonFinancials
+            .postCompanyAssociatedEuTaxonomyDataForNonFinancials(
+                companyAssociatedEuTaxonomyNonFinancialsData, false,
+            ).dataId
+
+        assertEquals(BackendQaStatus.pending, apiAccessor.metaDataControllerApi.getDataMetaInfo(dataId).qaStatus)
+
+        val qaServiceController = apiAccessor.qaServiceControllerApi
+        await().atMost(10, TimeUnit.SECONDS)
```

Comment:
Are the 10 seconds based on your observations? I would expect the dataId to be included in the unreviewed datasets a lot earlier.

Reply:
Me too. 10 seconds are just a sufficiently large upper bound. It does not wait 10 seconds if the dataset is listed earlier.

Reply:
Vorschlag:  Auf 1 Sekunde setzen
Hintergrund:  Wenn das im Test aus irgendeinem beliebigen Grund länger braucht, ist etwas schief und der Test sollte failen.

Wäre quasi eine Absicherung der Performance.  Was meinst du?

Reply:
Wir kamen zum Schluss dass wir 2 Sekunden setzen.
Der CI run ist ja mehr oder weniger ein best case, und mehr als 2 Sekunden erschienen uns seltsam.  Dann darf der Testa uch ruhig failen.

Reply:
Fixed